tools/io-trace-explorer/io_trace.cc

```diff
@@ -0,0 +1,415 @@
+            case 'D':
+                item.discard = true;
+                break;
+            }
+        }
+
+        /* Parse the offset and the length. */
+
+        item.offset = strtol(parts[7], &e, 10);
+        if (*e != '\0')
+            throw std::runtime_error(
+              std::string("Cannot parse the offset on line ") + std::to_string(lines));
+
+        if (strcmp(parts[8], "+") != 0)
+            throw std::runtime_error(
+              std::string("Unexpected payload on line ") + std::to_string(lines));
+
+        item.length = strtol(parts[9], &e, 10);
+        if (*e != '\0')
+            throw std::runtime_error(
+              std::string("Cannot parse the length on line ") + std::to_string(lines));
+
+        item.offset *= 512;
+        item.length *= 512;
+
+        /* Parse the other action-specfic data. */
+
+        int index = 10;
+        p = parts[index];
+
+        if (index < count && *p == '(') {
+            p++;
+            if (*p == '\0')
+                p = parts[++index];
+            if (index < count) {
+                p[strlen(p) - 1] = '\0';
+                item.duration = strtol(parts[index], &e, 10) / 1.0e6;
+                if (item.duration > 1e3)
+                    item.duration = 0;
+                p = parts[++index];
+            }
+        }
+
+        if (index < count && *p == '[') {
+            p++;
+            p[strlen(p) - 1] = '\0';
+            e = strrchr(p, '/');
+            if (e != NULL && isdigit(e[1]))
+                *e = '\0';
+            strncpy(item.process, p, sizeof(item.process));
+            p = parts[++index];
+        }
+
+        /* Right now we only care about reads and writes. */
+
+        if (!item.read && !item.write) {
+            continue;
+        }
+
+        /* Postprocess and save the line. */
+
+        add_data_point(device, io_trace_kind::DEVICE, item);
+    }
+}
+
+void
+io_trace_collection::load_from_file_wt_logs(FILE *f)
+{
+    /*
+     * Sample logs (with line breaks added):
+     *
+     * [1678914463:707985][29429:0x7f532741cc40], WT_SESSION.create: [WT_VERB_WRITE][DEBUG_2]:
+     *   write: /data/wt/test.wt, fd=10, offset=0, len=4096
+     * [1678914463:711029][29429:0x7f531effd640], log-server: [WT_VERB_WRITE][DEBUG_2]:
+     *    write: /data/wt/./WiredTigerTmplog.0000000003, fd=9, offset=0, len=128
+     * [1678914463:712302][29429:0x7f532741cc40], file:test.wt, WT_SESSION.create:
+     *   [WT_VERB_READ][DEBUG_2]: read: /data/wt/test.wt, fd=10, offset=0, len=4096
+     * [1678914463:713343][29429:0x7f532741cc40], file:WiredTiger.wt, WT_SESSION.create:
+     *   [WT_VERB_WRITE][DEBUG_2]: write: /data/wt/./WiredTigerLog.0000000001, fd=7,
+     * offset=1920, len=2048 [1678914463:715599][29429:0x7f53157fa640]:
+     * [WT_VERB_WRITE][DEBUG_2]: write: /data/wt/./WiredTigerLog.0000000001, fd=7, offset=3968,
+     * len=131328
+     */
+
+    char buf[256];
+    double first_timestamp = 0;
+    unsigned lines = 0;
```

Comment:
I think we prefer `unsigned int` in our codebase (I assume that was the intention).